index.bs

```diff
@@ -692,14 +692,33 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. For each |authenticator| currently available on this platform, perform the following steps:
 
-    1. Let |credentialList| be a new [=list=].
-
-    1. If <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code> [=list/is not empty=], execute a 
-        platform-specific procedure to determine which, if any, credentials in
-        <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code> are present on this |authenticator| by matching with 
-        <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}.{{ScopedCredentialDescriptor/id}}</code> and 
-        <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}.{{ScopedCredentialDescriptor/type}}</code>, and set |credentialList| to
-        this filtered list.
+    <span class="note">Note: the procedure to determine the [=authenticators=] present on a platform is 
+        platform-specific.</span>
+
+    1. Let |credentialList| be a new [=ordered set=].
+
+    1. If <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
+        <dl class="switch">
+            : [=list/is not empty=]
+            :: [=set/For each=] credential |C| in <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
+                1. If there is any credential |X| present on this |authenticator| that:
+                    - is mapped to |rpId|, and,
+                    - matches with <code>|C|.{{ScopedCredentialDescriptor/id}}</code> and 
+                        <code>|C|.{{ScopedCredentialDescriptor/type}}</code>,
+
+                    then [=set/append=] |X| to |credentialList|.
```

Comment:
Semantically, a [`ScopedCredentialDescriptor`](https://w3c.github.io/webauthn/#dictdef-scopedcredentialdescriptor) with no transports is equivalent to one that lists [all three](https://w3c.github.io/webauthn/#enumdef-transport), right? Remember to deal with that in the text.

Is it possible for an authenticator to be attached via more than one transport? Step 18 in [[[Create]]](https://w3c.github.io/webauthn/#createCredential) seems to assume not. Maybe the client would just treat it as two separate authenticators in that case? If we can assume that an authenticator has exactly one transport, then maybe you can replace 
> For each credential C in scopedOptions.[allowList](https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/w3c/webauthn/jeffh-empty-allowlist-387/index.bs#dom-scopedcredentialrequestoptions-allowlist):

with
> For each credential C in scopedOptions.[allowList](https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/w3c/webauthn/jeffh-empty-allowlist-387/index.bs#dom-scopedcredentialrequestoptions-allowlist) such that either C.[transports](https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/w3c/webauthn/jeffh-empty-allowlist-387/index.bs#dom-scopedcredentialdescriptor-transports) is empty or *authenticator*'s transport appears in C.[transports](https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/w3c/webauthn/jeffh-empty-allowlist-387/index.bs#dom-scopedcredentialdescriptor-transports):

And then the iteration over *credentialList* could be replaced by "Invoke the authenticatorGetAssertion operation on authenticator, with rpId, clientDataHash, credentialList, and clientExtensions as parameters."

@kpaulh @AngeloKai what are your implementations actually doing here? Do you deal with authenticators that have multiple transports and `allowList`s that constrain their credentials' transports?